### PR TITLE
Migration to web3modal API

### PR DESCRIPTION
**Pull request by quetool**
- Migration to web3modal API https://specs.walletconnect.com/2.0/specs/meta-clients/web3modal/api rather than using Explorer API and refactor of some logic
- Added pagination and loading state
- Bug fixes and enhancements

### CODE REVIEW

File: lib/utils/core/core_utils.dart [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    }
+
+    String encodedWcUrl = Uri.encodeComponent(wcUri);
+    W3MLoggerUtil.logger.i('Encoded WC URL: $encodedWcUrl');
+
+    return Uri.parse('${plainAppUrl}wc?uri=$encodedWcUrl');
+  }
+
+  @override
+  String getUserAgent() {
```
Comment (ignaciosantise) [inline lib/utils/core/core_utils.dart:69]: can you validate with diego this userAgent?

They currently accept `<sdk_version>/<core_version>/<os_version>`
But as you don't have a core version, check if you need to send `<sdk_version>/undefined/<os_version>` or you can just remove that part

File: lib/widgets/miscellaneous/responsive_container.dart [hunk 2]
```diff
@@ -97,18 +97,13 @@ class ResponsiveData extends InheritedWidget {
     return isPortraitMode ? kGridAxisCountP : kGridAxisCountL;
   }
 
-  static bool _isTabletSize(BuildContext context) {
-    final data = MediaQueryData.fromView(View.of(context));
-    return data.size.shortestSide < 600 ? false : true;
-  }
+  // static bool _isTabletSize(BuildContext context) {
+  //   final data = MediaQueryData.fromView(View.of(context));
+  //   return data.size.shortestSide < 600 ? false : true;
```
Comment (ignaciosantise) [inline lib/widgets/miscellaneous/responsive_container.dart:102]: is this comment needed?

**Comment by ignaciosantise** [comment]
@quetool i think you also need to update flutter docs right? with `featuredWalletIds` change

**Comment by quetool** [comment]
> @quetool i think you also need to update flutter docs right? with `featuredWalletIds` change

Yep, I'll do it tomorrow. 👌

File: lib/services/w3m_service/w3m_service.dart [hunk 3]
```diff
@@ -698,8 +699,8 @@ class W3MService with ChangeNotifier implements IW3MService {
     // Combine the redirect data from the session and the explorer API.
     // The explorer API is the source of truth.
     return Redirect(
-      native: explorerRedirect?.native ?? sessionRedirect?.native,
-      universal: explorerRedirect?.universal ?? sessionRedirect?.universal,
+      native: explorerRedirect ?? sessionRedirect?.native,
+      universal: explorerRedirect ?? sessionRedirect?.universal,
```
[the remark this answers is not in the document]
Comment (quetool) [reply lib/services/w3m_service/w3m_service.dart:703]: That is correct but that Redirect object is coming from the core sdk so I had to pass it universal as well. I'm planning to refactor that part later.

**Pull request merged**